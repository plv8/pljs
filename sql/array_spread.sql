-- ten seconds should be enough to show this doesn't destroy memory
set statement_timeout = '5s';
set pljs.memory_limit = '256';

do $$ Object.prototype [Symbol.iterator] = function() { return { next:() => this } };
[...({})];
$$ language pljs;

do $$ Object.prototype [Symbol.iterator] = function() { return { next:() => this } };
[...({})];
$$ language pljs;