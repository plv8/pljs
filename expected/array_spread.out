-- ten seconds should be enough to show this doesn't destroy memory
set statement_timeout = '5s';
set pljs.memory_limit = '256';
do $$ Object.prototype [Symbol.iterator] = function() { return { next:() => this } };
[...({})];
$$ language pljs;
ERROR:  execution error
DETAIL:  InternalError: out of memory
    at <anonymous> (<function>:3)
    at <eval> (<function>:5)

do $$ Object.prototype [Symbol.iterator] = function() { return { next:() => this } };
[...({})];
$$ language pljs;
ERROR:  execution error
DETAIL:  InternalError: out of memory
    at <anonymous> (<function>:3)
    at <eval> (<function>:5)

